src/metabase/lib/metadata/calculation.cljc

```diff
@@ -551,52 +565,35 @@
   [:sequential ::visible-column])
 
 (mr/def ::visible-columns.options
-  "Schema for options passed to [[visible-columns]] and [[visible-columns-method]]."
+  "Schema for options passed to [[visible-columns]]."
   [:merge
    [:ref ::returned-columns.options]
    [:map
-    ;; these all default to true
     [:include-joined?                              {:optional true, :default true} :boolean]
     [:include-expressions?                         {:optional true, :default true} :boolean]
     [:include-implicitly-joinable?                 {:optional true, :default true} :boolean]
     [:include-implicitly-joinable-for-source-card? {:optional true, :default true} :boolean]]])
 
+(def ^:private visible-columns-options-keys
+  (mu/map-schema-keys ::visible-columns.options))
+
 (mu/defn- default-visible-columns-options :- ::visible-columns.options
   []
   {:include-joined?                              true
    :include-expressions?                         true
    :include-implicitly-joinable?                 true
    :include-implicitly-joinable-for-source-card? true})
 
-(defmulti visible-columns-method
-  "Impl for [[visible-columns]]."
-  {:arglists '([query stage-number x options])}
-  (fn [_query _stage-number x _options]
-    (lib.dispatch/dispatch-value x))
-  :hierarchy lib.hierarchy/hierarchy)
-
-(defmethod visible-columns-method :dispatch-type/nil
-  [_query _stage-number _x _options]
-  [])
-
-;;; default impl is just the impl for [[returned-columns-method]]
-(defmethod visible-columns-method :default
-  [query stage-number x options]
-  (returned-columns-method query stage-number x options))
-
-;;; if you pass in an integer assume it's a stage number; use the method for the query stage itself.
-(defmethod visible-columns-method :dispatch-type/integer
-  [query _stage-number stage-number options]
-  (visible-columns-method query stage-number (lib.util/query-stage query stage-number) options))
-
-;; TODO (Cam 8/7/25) -- Braden and I are in agreement that `visible-columns` fundamentally does not make sense for
-;; anything other than a specific query stage, and we should probably remove it entirely in the near future and leave
-;; just the stage implementation. See https://github.com/metabase/metabase/pull/61767#discussion_r2260558780
+;;; TODO (Cam 8/7/25) -- historically `visible-columns` worked on a bunch of stuff besides just a stage, but Braden
+;;; pointed out that it really only makes any sense at all for a stage here
+;;; https://metaboat.slack.com/archives/C0645JP1W81/p1754587431585559 . For historic reasons the main entrypoint to
+;;; the function still lives here, but it's now just a shim that calls [[metabase.lib.stage/visible-columns*]] -- we
+;;; should combine these so there's no more indirection. It should probably live in [[metabase.lib.stage]] going
+;;; forward, but we'd need to update a lot of code that calls this version of [[visible-columns]].
```

Comment:
I think this is out of date? It seems like you called the inner function `metabase.lib.stage/-visible-columns`.